### PR TITLE
669 view change status

### PR DESCRIPTION
**Pull request by marcellmueller**
- feat: add column change_reason to application_status table
- feat: add permissions and new status types
- feat: add createApplicationStatus mutation
- feat: add change status menu and modal
- test: add change status tests
- feat: set status pill width
- feat: add rls policies for application_status
- fix: handle change status for withdrawn application

Implements # \<issue number\>

### CODE REVIEW

File: app/components/Analyst/ChangeStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  appearance: none;
+  padding: 6px 12px;
+  height: 30px;
+  margin-bottom: 16px;
+  width: ${(props) => props.statusStyles?.pillWidth};
+  background: ${(props) => props.statusStyles?.backgroundColor}
+    url("data:image/svg+xml;utf8,<svg viewBox='0 0 140 140' width='24' height='24' xmlns='http://www.w3.org/2000/svg'>
+    <g><path d='m121.3,34.6c-1.6-1.6-4.2-1.6-5.8,0l-51,51.1-51.1-51.1c-1.6-1.6-4.2-1.6-5.8,0-1.6,1.6-1.6,4.2 0,5.8l53.9,53.9c0.8,0.8 1.8,1.2 2.9,1.2 1,0 2.1-0.4 2.9-1.2l53.9-53.9c1.7-1.6 1.7-4.2 0.1-5.8z' fill='${(
+      props
+    ) => props.statusStyles?.primary.replace('#', '%23')}'/></g></svg>")
```
[the remark this answers is not in the document]
Comment (marcellmueller) [reply app/components/Analyst/ChangeStatus.tsx:40]: False positive, we only want to replace the `#` in the colour hex code as a hash in a url starts a fragment identifier so we want to escape that so our svg fill colour will work.

File: app/components/Analyst/ChangeStatus.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  console.log(status);
+
+  const hiddenStatusTypes = ['draft', 'submitted', 'withdrawn'];
+
+  // Filter unwanted status types
+  const statusTypes = allApplicationStatusTypes.nodes.filter(
+    (statusType) => !hiddenStatusTypes.includes(statusType.name)
+  );
+
+  const getStatus = (statusName, statusList) => {
```
Comment (AntBush) [inline app/components/Analyst/ChangeStatus.tsx:85]: Nit: We can move this out of the definition of the component since it's a pure function

**Review by AntBush — APPROVED** [review]
Looks good! Just the small nit if you want to change it and the `console.log` on `ChangeStatus.tsx`

File: app/components/Analyst/ChangeStatus.tsx (hunk 2 above)
Comment (marcellmueller) [reply app/components/Analyst/ChangeStatus.tsx]: I update that thank you :bow:

**Pull request merged**